### PR TITLE
`make goimports` support Linux (make `sed` option OS-aware)

### PR DESCRIPTION
**Pull request by magodo**
Make the sed OS-aware in goimport-file.sh & run it

### CODE REVIEW

**Review by manicminer — DISMISSED** [review]
Thanks @magodo, this LGTM 👍

File: scripts/goimport-file.sh [hunk 1]
```diff
@@ -7,10 +7,20 @@ if [ $# != 1 ] ; then
...
     /^$/d
   }
-' $1
+'
+
+case "$OSTYPE" in
+    "linux-gnu"*)
+        sed -i -e "$sed_expression" $1
+        ;;
+    "darwin"*)
```
Comment (manicminer) [inline scripts/goimport-file.sh:21]: This does assume the version of sed available is the default on each platform - many users use gnu sed on darwin however. This should detect gnu vs non gnu:

```suggestion
sed_version="$(gsed --version 2>/dev/null)"
case "${sed_version}" in
    *gnu*)
        sed -i -e "$sed_expression" $1
        ;;
    *)
```

Comment (magodo) [reply scripts/goimport-file.sh:21]: Does `gsed` installation alias itself as `sed`?

Comment (manicminer) [reply scripts/goimport-file.sh:21]: It doesn't by default, but the brew package does give instructions on how to do so, and it's not an uncommon configuration

File: scripts/goimport-file.sh [hunk 2]
```diff
@@ -19,7 +19,11 @@ case "$OSTYPE" in
         sed -i -e "$sed_expression" $1
         ;;
     "darwin"*)
-        sed -i '' -e "$sed_expression" $1
+        if command -v gsed; then
+            gsed -i -e "$sed_expression" $1
+        else
+            sed -i '' -e "$sed_expression" $1
+        fi
```
Comment (manicminer) [inline scripts/goimport-file.sh:26]: It's not so much that we should use gsed if it's present, but that `sed` could be either GNU or posix sed and we should detect that.

Comment (magodo) [reply scripts/goimport-file.sh]: I've updated to check the `sed` version output to see whether it contains *GNU*. While I'm not sure whether the output of the POSIX `sed --version` also output text that will match *GNU*..

**Review by manicminer — APPROVED** [review]
LGTM

**Pull request merged**